### PR TITLE
New: (Indexer) Shizaproject

### PR DESCRIPTION
**Pull request by dm3ch**
#### Database Migration
NO

#### Description
Adding Shizaproject indexer support

#### Todos
Tests
Translation Keys (./src/NzbDrone.Core/Localization/Core/en.json)
[Wiki Updates](https://wiki.servarr.com)

#### Issues Fixed or Closed by this PR

* Fixes #XXXX
* Related https://github.com/Prowlarr/Prowlarr/issues/29

### CODE REVIEW

File: src/NzbDrone.Core/Indexers/Definitions/Shizaproject.cs [hunk 1]
```diff
@@ -0,0 +1,348 @@
...
+    public class ShizaprojectRequestGenerator : IIndexerRequestGenerator
+    {
+        public ShizaprojectSettings Settings { get; set; }
+        public IndexerCapabilities Capabilities { get; set; }
+
+        public ShizaprojectRequestGenerator()
+        {
+        }
+
+        private IEnumerable<IndexerRequest> GetPagedRequests(string term, int[] categories, string imdb = null)
```
Comment (Qstick) [inline src/NzbDrone.Core/Indexers/Definitions/Shizaproject.cs:79]: Ditch IMDb here if it’s never used

**Review by Qstick — CHANGES_REQUESTED** [review]
Just the one nitpick

**Comment by dm3ch** [comment]
@Qstick I have tested this indexer manually and found strange issue. Manual search works, but Sonarr search doesn't.
Maybe you have any ideas, before I would start deep debugging? :)

File: src/NzbDrone.Core/Indexers/Definitions/Shizaproject.cs (hunk 1 above)
Comment (dm3ch) [reply src/NzbDrone.Core/Indexers/Definitions/Shizaproject.cs]: Deleted

**Comment by Qstick** [comment]
Which categories is Sonarr using? Because only one is supported by this definition (TV Anime)

**Comment by dm3ch** [comment]
> Which categories is Sonarr using? Because only one is supported by this definition (TV Anime)

I tried to enable all available categories on sonarr side but it didn't help me. :(

**Comment by Qstick** [comment]
Jump on discord if you want to debug some, I’m not somewhere I can test the PR currently

**Comment by dm3ch** [comment]
@Qstick I have found what caused this issue.

During manual search I have "search term", but during automatic query it was "search term S01".
So I added regexp to filter that before requesting API.

The most interesting - when I was looking Prowlarr requests history, request results was 0, when I pressed manual search button in history view I got results. So search button from history view generates different search term than original request. :)

**Comment by bakerboy448** [comment]
update when merged https://requests.prowlarr.com/b/feedback/p/torrent-shizaproject

**Pull request merged**